react-native/react/actions/update.js

```diff
@@ -0,0 +1,105 @@
+/* @flow */
+
+import * as Constants from '../constants/update'
+import engine from '../engine'
+import moment from 'moment'
+
+import {updateUi} from '../constants/types/keybase_v1'
+import type {Update} from '../constants/types/flow-types'
+import type {ShowUpdateAction, RegisterUpdateListenerAction, OnCancelAction, OnSkipAction, OnSnoozeAction, OnUpdateAction} from '../constants/update'
+import type {Dispatch} from '../constants/types/flux'
+
+let updateResponse: ?(result: any/* UpdatePromptRes */) => void
```

Comment:
I have a comment about this. We don't fill in all the return values so we
can't use the type as is

On Wednesday, December 16, 2015, Marco Munizaga <notifications@github.com>
wrote:

> In react-native/react/actions/update.js
> <https://github.com/keybase/client/pull/1544#discussion_r47842901>:
>
> > @@ -0,0 +1,105 @@
> > +/* @flow */
> > +
> > +import * as Constants from '../constants/update'
> > +import engine from '../engine'
> > +import moment from 'moment'
> > +
> > +import {updateUi} from '../constants/types/keybase_v1'
> > +import type {Update} from '../constants/types/flow-types'
> > +import type {ShowUpdateAction, RegisterUpdateListenerAction, OnCancelAction, OnSkipAction, OnSnoozeAction, OnUpdateAction} from '../constants/update'
> > +import type {Dispatch} from '../constants/types/flux'
> > +
> > +let updateResponse: ?(result: any/* UpdatePromptRes */) => void
>
> I think this should be:
> let updateResponse: {result: (payload: any) => void}
>
> also can we type UpdatePromptRes?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/keybase/client/pull/1544/files#r47842901>.
>